sdk/tables/azure-data-tables/src/test/java/com/azure/data/tables/AzureTableImplTest.java

```diff
@@ -0,0 +1,519 @@
+            requestId, ResponseFormat.RETURN_CONTENT, null, Context.NONE))
+            .assertNext(response -> {
+                Assertions.assertEquals(expectedStatusCode, response.getStatusCode());
+            })
+            .expectComplete()
+            .verify();
+    }
+
+    @Test
+    void createTableDuplicateName() {
+        // Arrange
+        String tableName = testResourceNamer.randomName("test", 20);
+        TableProperties tableProperties = new TableProperties().setTableName(tableName);
+        createTable(tableName);
+        String requestId = testResourceNamer.randomUuid();
+
+        // Act & Assert
+        StepVerifier.create(azureTable.getTables().createWithResponseAsync(tableProperties,
+            requestId, ResponseFormat.RETURN_CONTENT, null, Context.NONE))
+            .expectError(com.azure.data.tables.implementation.models.TableServiceErrorException.class)
+            .verify();
+    }
+
+    @Test
+    void deleteTable() {
+        // Arrange
+        String tableName = testResourceNamer.randomName("test", 20);
+        createTable(tableName);
+        int expectedStatusCode = 204;
+        String requestId = testResourceNamer.randomUuid();
+
+        // Act & Assert
+        StepVerifier.create(azureTable.getTables().deleteWithResponseAsync(tableName, requestId,
+            Context.NONE))
+            .assertNext(response -> {
+                Assertions.assertEquals(expectedStatusCode, response.getStatusCode());
+            })
+            .expectComplete()
+            .verify();
+    }
+
+    @Test
+    void deleteNonExistentTable() {
+        // Arrange
+        String tableName = testResourceNamer.randomName("test", 20);
+        String requestId = testResourceNamer.randomUuid();
+
+        // Act & Assert
+        StepVerifier.create(azureTable.getTables().deleteWithResponseAsync(tableName, requestId,
+            Context.NONE))
+            .expectError(com.azure.data.tables.implementation.models.TableServiceErrorException.class)
+            .verify();
+    }
+
+    @Test
+    void queryTable() {
+        // Arrange
+        QueryOptions queryOptions = new QueryOptions()
+            .setFormat(OdataMetadataFormat.APPLICATION_JSON_ODATA_MINIMALMETADATA);
+        String tableA = testResourceNamer.randomName("AtestA", 20);
+        String tableB = testResourceNamer.randomName("BtestB", 20);
+        createTable(tableA);
+        createTable(tableB);
+        int expectedStatusCode = 200;
+        String requestId = testResourceNamer.randomUuid();
+
+        // Act & Assert
+        StepVerifier.create(azureTable.getTables().queryWithResponseAsync(requestId, null,
+            queryOptions, Context.NONE))
+            .assertNext(response -> {
+                Assertions.assertEquals(expectedStatusCode, response.getStatusCode());
+                Assertions.assertEquals(response.getValue().getValue().get(0).getTableName(), tableA);
```

Comment:
Adding a couple of assertions may prevent null pointer exceptions and out of bounds exceptions (if there are not 2 elements in the table).
```java
assertNotNull(response.getValue(), "Expected there to be a result.");

var results = response.getValue().getValue();

assertNotNull(results, "Expected there to be a set of items.");
assertEquals(2, results.getSize());

//... your other assertions.
```

Reply:
Same with any other instances where you do this .getValue().getValue().